src/components/structures/MatrixChat.js

```diff
@@ -304,6 +305,9 @@ module.exports = React.createClass({
                 });
                 break;
             case 'view_room':
+                // by default we autoPeek rooms, unless we were called explicitly with
+                // autoPeek=false by something like RoomDirectory who has already peeked
+                this.setState({ autoPeek : payload.auto_peek === false ? false : true });
```

Comment:
OH! I understand now. I would c/p that exactly and use that because that is by far the clearer syntax even if it isn't as terse.

Now that I know this, your original line works as you intend it to, it's just I didn't understand your intent from the code. As an aside, `if undefined default true` is generally hard to do clearly and succinctly in Javascript and usually ends up being *the single use case* for `==`:

```javascript
if (someVar == null) { // returns true for undefined and null, but false for the literal false and 0
  someVar = true;
}
// do something with someVar